SAMtags.tex

```diff
@@ -453,6 +455,112 @@ \subsubsection{Color space}
 Color read quality on the original strand of the read. Same encoding as {\sf QUAL}; same length as {\tt CS}.
 \end{description}
 
+\subsection{Base modifications}
+
+Base modifications, including base methylation, are represented as a
+series of edits from the primary unmodified sequence stored in the
+main SAM {\sf SEQ} field.  If the modifed base has no natural
+unmodified form then this should be stored as ``N''.
+
+Each modified base listed also has a quality value associated with
+it.  Given the unmodified base already has a phred likelihood, this
+base modification quality should be interpreted as the likelihood of
+this modification being correct, rather than the base being unmodified.
+
+\begin{description}
+\item[MM:Z:\tagregex{([ACGTN][-+][a-z](,[0-9]+)*;)*}]
+\hfill\\
+The first character is the unmodified base as seen in the {\sf SEQ}
+field, one of {\tt A}, {\tt C}, {\tt G}, {\tt T} or {\tt N}, with
+the exception that {\tt N} is used to match any base rather than
+strictly {\tt N}.  This is followed by either plus or minus
+indicating the strand the modification was observed on (relative to
+the original sequenced strand of {\sf SEQ} with plus meaning same
+orientation\footnote{Hence a tool that may reverse complement sequences does not need to understand how to manipulate the {\tt MM} and {\tt MP} tags.}), and a base modification symbol.
+This is then followed by a comma separated list of how many unmodified seq bases of the
+stated base type to skip, stored as a delta to the last and starting
+with 0 as the first (or next) base.  Hence this number series is
+comparable to the numbers in an {\tt MD} tag.
+
+For example {\tt C+m,5,12,0;} tells us there are three 5-Methylcytosine
+bases in the original {\sf SEQ}.  The first 5 {\tt C} bases are
+unmodified and the 6th is modified, as are the 19th (12 between the
+6th and 19th) and 20th.  Similarly {\tt G-m,14;} indicates the 15th
+{\tt G} is a 5-Methylcytosine on the opposite strand.
+
+This permits modifications to be listed on either strand with the rare
+potential for both strands to have a modification at the same site.
+
+Note it is permitted for the coordinate list to be empty (for example {\tt MM:Z:C+m;}), which may be used as an explicit indicator that this base modification is not present.
```

Comment:
I don't think any can right now, but there were (are?) technologies that unwound the double stranded DNA and sequenced one strand followed by the other strand.   IIRC PacBio SmartBells did this at one point, or maybe still do, and ONT earlier had their "2D" reads which did this.

The point being the format has to cope with what may come along down the road and this is conceivable, albeit a rather unlikely event to need to encode.

Reply:
ONT has had 1D^2 ever since they stopped 2D, from which the forward and reverse-complement reads from a single double-stranded sequence (as separate reads) can be combined into a single sequence. They also plan to re-introduce fully-connected 2D sequences in the future.